### PR TITLE
DOC: Fix What's New for 7.0 and bad intersphinx, remove astropy-dev tag

### LINKED ISSUE #11383: DOC: Change remote name to upstream
There were some discussions on astropy/astropy-project#151 regarding usage of `origin` vs `upstream` (vs `astropy`) as remote name in examples. As @taldcroft pointed out at https://github.com/astropy/astropy-project/issues/151#issuecomment-795924703 and https://github.com/astropy/astropy-project/issues/151#issuecomment-795927271 , `upstream` is the most widely used term, so perhaps it is less confusing if we update our dev docs to use `upstream`.

Example doc (one of many) that is affected: `docs/development/workflow/git_edit_workflow_examples.rst`

@embray , perhaps this can be part of your campaign to improve the dev docs?

### LINKED ISSUE #16531: Ensure doc builds are selfcontained, not relying on astropy-dev
### What is the problem this feature will solve?

Currently, if the docs updates cause some references to disappear, the readthedocs build does not fail, as it gets references from intersphinx mappings: https://github.com/astropy/astropy/blob/fe7b649b433c057f1fa4ca638b60e0b7097a024d/docs/conf.py#L112-L114

This now has meant several times after a PR is merged, `main` will fail and there has to be a follow-up PR (most recently, #16513, followed up by #16530).

Can we avoid this? Is that line in `conf.py` for astropy-dev in fact needed?

### Describe the desired outcome

Tests ensure that when a PR is merger, main continues to pass CI.

### Additional context

_No response_

### LINKED ISSUE #16626: DOC: What's New TOC missing for v7.0.0
I see contents but the TOC is empty. Also see https://github.com/astropy/astropy/pull/16561#discussion_r1657462892

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to add missing intersphinx for what's new entries and also fix version typo in the text. ~Should be uncontroversial enough that I might just merge without review once the doc builds but if you want to review before then, feel free.~

Turns out I also have to fix other things as a follow-up of #16561, so probably needs a proper review now.

* Fixes #16626 
* Fixes #16531
* Fixes #11383
* Fixes the RTD warnings that is causing `latest` build to fail (e.g., https://readthedocs.org/projects/astropy/builds/24870868/)

I added `Extra CI` for linkcheck only.

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by pllim** [comment]
Wut...? 

```
docs/development/git_resources.rst:139: WARNING: undefined label: 'fork_a_copy' [ref.ref]
docs/development/git_resources.rst:182: WARNING: undefined label: 'fetch-latest' [ref.ref]
docs/development/maintainers/maintainer_workflow.rst:98: WARNING: undefined label: 'set_upstream_main' [ref.ref]
docs/development/maintainers/testhelpers.rst:86: WARNING: py:obj reference target not found: astropy.test [ref.obj]
docs/development/maintainers/testhelpers.rst:89: WARNING: py:obj reference target not found: astropy.test [ref.obj]
```

Maybe same problem as https://github.com/astropy/astropy/issues/16531

File: docs/development/forking_button.png
Comment (pllim) [inline docs/development/forking_button.png:1]: This is no longer used anywhere. I think this not getting deleted back in #16561 was an oversight.

Comment (taldcroft) [reply docs/development/forking_button.png:1]: Yup!

**Review by taldcroft — APPROVED** [review]
@pllim - this looks great, thanks for the clean-up and catching a few issues from #16561!

I looked through the diffs and checked the RTD output briefly and it looks good.

**Comment by pllim** [comment]
Thanks for the quick review!

**Pull request merged**